packages/@guardian/source-react-components-development-kitchen/src/footer-with-contents/FooterWithContents.tsx

```diff
@@ -0,0 +1,85 @@
+import { css } from '@emotion/react';
+import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import {
+	brand,
+	from,
+	neutral,
+	space,
+	textSans,
+} from '@guardian/source-foundations';
+import { Container } from '@guardian/source-react-components';
+import { BackToTop } from '@guardian/source-react-components/src/footer/BackToTop';
+import type { ReactNode } from 'react';
+
+const footer = css`
+	background-color: ${brand[400]};
+	color: ${neutral[100]};
+	padding-bottom: ${space[1]}px;
+	${textSans.medium()};
+`;
+
+const contentWrapperStyles = css`
+	display: flex;
+	position: relative;
+`;
+
+const copyrightStyles = css`
+	${textSans.xxsmall()};
+	padding-top: ${space[6]}px !important;
+	padding-bottom: 18px !important;
+	${from.tablet} {
+		padding-top: ${space[3]}px !important;
+	}
+`;
+
+const backToTopStyles = css`
+	background-color: ${brand[400]};
+	padding: 0 ${space[2]}px;
+	position: absolute;
+	bottom: -21px;
+	right: ${space[3]}px;
+	${from.tablet} {
+		padding: 0 5px;
+		right: 0;
+	}
+`;
+
+export interface FooterWithContentsProps {
+	children: ReactNode;
+	/**
+	 * A container component that matches the type signature of the Source Container.
+	 * Use where different internal padding or other behaviour needs to be applied to match an existing layout.
+	 */
+	container?: typeof Container;
+}
+
+export const FooterWithContents = ({
+	children,
+	container = Container,
+}: FooterWithContentsProps): EmotionJSX.Element => {
+	const ContainerComponent = container;
```

Comment:
could `container` maybe just be called `ContainerComponent` at the function argument level?

Reply:
Agree - I think both this and line 58 are a little confusing

Reply:
Yeah, that'd be a lot clearer!